File: AV1_C206_L4.cs
using System.Collections.Generic;
using System;

namespace ConsoleApp;

static public class AV1_C206_L4 {
    static public void Main() {
        Estante estanteBraba = new Estante(idEstante: 666, letra: 'a');

        while (true) {
            Console.WriteLine("Seleciona uma opção ae: \n" +
                              "1. Add livro\n" +
                              "2. Mostrar info estantes e livros\n" +
                              "3. Mostrar porcentagem de livros em cada genero\n" +
                              "4. Mostrar info do livro e autor com mais e menos paginas\n" +
                              "5. Sair");
            int userInput = Convert.ToInt32(Console.ReadLine());

            switch (userInput) {
                case 1:
                    Console.Write("Titulo: ");
                    string titulo = Console.ReadLine() ?? string.Empty;
                    Console.Write("Genero literario: ");
                    string genLiterario = Console.ReadLine() ?? string.Empty;
                    Console.Write("Numero de folhas: ");
                    int folhas = Convert.ToInt32(Console.ReadLine());
                    Console.Write("Editora: ");
                    string editora = Console.ReadLine() ?? string.Empty;

                    Console.Write("nomeAutor: ");
                    string nomeAutor = Console.ReadLine() ?? string.Empty;
                    Console.Write("Ano de nascimento do autor: ");
                    int anoNascimentoAutor = Convert.ToInt32(Console.ReadLine());
                    Console.Write("Profissao do autor: ");
                    string profissaoAutor = Console.ReadLine() ?? string.Empty;


                    Livro novoLivro = new Livro(
                        titulo: titulo,
                        genLiterario: genLiterario,
                        folhas: folhas,
                        editora: editora,
                        autorNome: nomeAutor,
                        AutorAnoNascimento: anoNascimentoAutor,
                        autorProfissao: profissaoAutor
                    );

                    estanteBraba.addLivros(novoLivro);
                    break;
                case 2:
                    estanteBraba.mostrarInfos();
                    break;
                case 3:
                    estanteBraba.porcentagemGen();
                    break;
                case 4:
                    estanteBraba.livroMaisEMenos();
                    break;
                case 5:
                    return;
                default:
                    Console.WriteLine("?");
                    break;
            }
        }
    }


    public class Estante {
        private int idEstante;
        private char letra;
        private List<Livro> livros = new List<Livro>();


        public Estante(int idEstante, char letra) {
            this.idEstante = idEstante;
            this.letra = letra;
        }

        public void addLivros(Livro livro) {
            livros.Add(livro);
            Console.WriteLine("Livro adicionado.");
        }

        public void porcentagemGen() {
            int suspense = 0;
            int drama = 0;

            foreach (var livro in livros) {
                if (livro.genLiterario.ToLower() == "suspense")
                    suspense++;
                else if (livro.genLiterario.ToLower() == "drama")
                    drama++;
            }

            int totalLivros = suspense + drama;
            if (totalLivros is not 0) {
                Console.WriteLine($"% suspense {suspense * 100 / totalLivros}");
                Console.WriteLine($"% drama {drama * 100 / totalLivros}");
            }
        }

        public void livroMaisEMenos() {
            int maisPaginas = 0;
            int menosPaginas = Int32.MaxValue;
            Livro livromaisPaginas = null;
            Livro livromenosPaginas = null;

            foreach (var livro in livros) {
                if (livro.folhas > maisPaginas) {
                    maisPaginas = livro.folhas;
                    livromaisPaginas = livro;
                }

                if (livro.folhas < menosPaginas) {
                    menosPaginas = livro.folhas;
                    livromenosPaginas = livro;
                }
            }

            Console.WriteLine("Mais paginas: ");
            livromaisPaginas.mostrarInfos();
            Console.WriteLine("Menos paginas: ");
            livromenosPaginas.mostrarInfos();
        }

        public void mostrarInfos() {
            Console.WriteLine(idEstante + " " + letra);
            foreach (var livro in livros) {
                livro.mostrarInfos();
            }
        }
    }


    public class Livro {
        private string titulo;
        public string genLiterario;
        public int folhas;
        private string editora;
        private Autor autor;


        public Livro(string titulo, string genLiterario, int folhas, string editora, string autorNome,
            int AutorAnoNascimento, string autorProfissao) {
            this.titulo = titulo;
            this.genLiterario = genLiterario;
            this.folhas = folhas;
            this.editora = editora;
            this.autor = new Autor(nome: autorNome, anoNascimento: AutorAnoNascimento, profissao: autorProfissao);
        }

        public void mostrarInfos() {
            Console.WriteLine(this.titulo + " " + this.genLiterario + " " + this.folhas + " " + this.editora);
        }
    }


    public class Autor {
        private string nome;
        private int anoNascimento;
        private string profissao;


        public Autor(string nome, int anoNascimento, string profissao) {
            this.nome = nome;
            this.anoNascimento = anoNascimento;
            this.profissao = profissao;
        }

        public void mostrarInfos() {
            Console.WriteLine(this.nome + " " + this.anoNascimento + " " + this.profissao);
        }
    }
}

File: AV1_C206_L4.java
import java.util.Scanner;


public class AV1_C206_L4 {
    public static void main(String[] args) {
        Estante estanteBraba = new Estante();
        estanteBraba.idEstante = 666;
        estanteBraba.letra = 'c';

        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("Seleciona uma opcao ae: \n" +
                    "1. Add livro\n" +
                    "2. Mostrar info estantes e livros\n" +
                    "3. Mostrar porcentagem de livros em cada genero\n" +
                    "4. Mostrar info do livro e autor com mais e menos paginas\n" +
                    "5. Sair");
            int userInput = sc.nextInt();

            switch (userInput) {
                case 1:
                    Livro novoLivro = new Livro();

                    System.out.print("Titulo: ");
                    novoLivro.titulo = sc.next();

                    System.out.print("Genero literario: ");
                    novoLivro.genLiterario = sc.next();

                    System.out.print("Numero de folhas: ");
                    novoLivro.qtdFolhas = sc.nextInt();

                    System.out.print("Editora: ");
                    novoLivro.editora = sc.next();

                    System.out.print("nomeAutor: ");
                    novoLivro.autor.nome = sc.next();

                    System.out.print("Ano de nascimento do autor: ");
                    novoLivro.autor.anoNascimento = sc.nextInt();

                    System.out.print("Profissao do autor: ");
                    novoLivro.autor.profissao = sc.next();

                    estanteBraba.addLivros(novoLivro);
                    break;
                case 2:
                    estanteBraba.mostrarInfos();
                    break;
                case 3:
                    estanteBraba.porcentagemGen();
                    break;
                case 4:
                    estanteBraba.livroMaisEMenos();
                    break;
                case 5:
                    return;
                default:
                    System.out.println("?");
                    break;
            }
        }
    }


    public static class Estante {
        private int idEstante;
        private char letra;
        private Livro[] livros = new Livro[30];


        public void addLivros(Livro livro) {
            for (int i = 0; i < livros.length; i++) {
                if (livros[i] == null) {
                    livros[i] = livro;
                    System.out.println("\nLivro adicionado.\n");
                    return;
                }
            }
            System.out.println("\nPrateleira cheia, nao e possivel adicionar mais livros.\n");
        }

        public void porcentagemGen() {
            int suspense = 0;
            int drama = 0;

            for (Livro livro : livros) {
                if(livro == null)
                    continue;
                if (livro.genLiterario.toLowerCase().equals("suspense"))
                    suspense++;
                else if (livro.genLiterario.toLowerCase().equals("drama"))
                    drama++;
            }

            int totalLivros = suspense + drama;
            if (totalLivros != 0) {
                System.out.println("Suspense " + (float)(suspense * 100 / totalLivros) + "%");
                System.out.println("Drama " + (float)(drama * 100 / totalLivros) + "%");
            }
            System.out.println();
        }

        public void livroMaisEMenos() {
            int maisFolhas = 0;
            int menosFolhas = Integer.MAX_VALUE;
            Livro livromaisPaginas = null;
            Livro livromenosPaginas = null;

            for (Livro livro : livros) {
                if(livro == null)
                    continue;

                if (livro.qtdFolhas > maisFolhas) {
                    maisFolhas = livro.qtdFolhas;
                    livromaisPaginas = livro;
                }
                if (livro.qtdFolhas < menosFolhas) {
                    menosFolhas = livro.qtdFolhas;
                    livromenosPaginas = livro;
                }
            }
            
            System.out.println();
            
            System.out.println("Mais paginas: ");
            if (livromaisPaginas != null) 
            livromaisPaginas.mostrarInfos();
            else
            System.out.println("Nao possui livro.");
            
            System.out.println();

            System.out.println("Menos paginas: ");
            if (livromenosPaginas != null)
                livromenosPaginas.mostrarInfos();
            else
                System.out.println("Nao possui livro.");
            
            System.out.println();
        }

        public void mostrarInfos() {
            System.out.println("\n" + idEstante + " " + letra);
            for (Livro livro : livros) {
                if(livro != null)
                    livro.mostrarInfos();
            }
            System.out.println();
        }
    }


    public static class Livro {
        private String titulo;
        public String genLiterario;
        public int qtdFolhas;
        public String editora;
        private Autor autor;


        public Livro() {
            this.autor = new Autor();
        }

        public void mostrarInfos() {
            System.out.println(this.titulo + " " + this.genLiterario + " " + this.qtdFolhas);
        }
    }


    public static class Autor {
        private String nome;
        private int anoNascimento;
        private String profissao;


        public void mostrarInfos() {
            System.out.println(this.nome + " " + this.anoNascimento + " " + this.profissao);
        }
    }

}
